### PR TITLE
Rayoffiah/adding stem support

### PR DESCRIPTION
**Pull request by RayOffiah**
Adding STEM support to the site.

Not working at the moment; does not appear to be seeing asciidoc.stem setting.

### CODE REVIEW

**Comment by mojavelinux** [comment]
The preview doesn't provide a full Antora environment. Much of the UI model, and the helpers in particular, are mocked. Therefore, it's not going to provide the depth of metadata needed for this check. But it will work if you try it on a real site.

See https://github.com/couchbase/docs-ui/blob/master/gulp.d/tasks/build-preview-pages.js#L313-L315

File: src/partials/footer-scripts.hbs [hunk 1]
```diff
@@ -17,6 +17,28 @@
 <script defer src="{{#unless preview}}{{{uiRootPath}}}/js/vendor/{{/unless}}fontawesome-icon-defs.js"></script>
 <script defer src="{{{uiRootPath}}}/js/vendor/fontawesome.js"></script>
 <script async src="{{{uiRootPath}}}/js/vendor/highlight.js"></script>
+
```
Comment (mojavelinux) [inline src/partials/footer-scripts.hbs]: Remove this empty line.

File: src/partials/footer-scripts.hbs [hunk 2]
```diff
@@ -17,6 +17,28 @@
...
+{{#with (resolvePage page.relativeSrcPath model=false)}}
+    {{#unless (eq ./asciidoc.attributes.stem undefined)}}
```
Comment (mojavelinux) [inline src/partials/footer-scripts.hbs]: The convention we've been following is to not indent logic tags. It creates superfluous nesting in the resulting HTML.

File: src/partials/footer-scripts.hbs [hunk 3]
```diff
@@ -17,6 +17,28 @@
...
+        <script type="text/x-mathjax-config">
```
Comment (mojavelinux) [inline src/partials/footer-scripts.hbs]: The convention we've been following is to not indent top-level tags in a template, such as `<script>`.

**Review by mojavelinux — CHANGES_REQUESTED** [review]

**Review by mojavelinux — APPROVED** [review]

**Review by maria-robobug — APPROVED** [review]
👍

**Comment by mojavelinux** [comment]
FYI, you can obtain the UI bundle from a deploy preview in any Antora environment by appending `dist/ui-bundle.zip` to the deploy preview URL. For example:

https://deploy-preview-127--cb-docs-ui.netlify.app/dist/ui-bundle.zip

You can either set that value in the playbook or pass it to Antora using the `--ui-bundle-url` CLI option.

This allows you to test a change to the UI with the real content before merging into the mainline.

**Comment by RayOffiah** [comment]
I didn’t think of the command line option. Thanks; I’ll remember that one.

From: Dan Allen ***@***.***>
Date: Monday, 15 November 2021 at 09:52
To: couchbase/docs-ui ***@***.***>
Cc: Ray Offiah ***@***.***>, Author ***@***.***>
Subject: Re: [couchbase/docs-ui] Rayoffiah/adding stem support (PR #127)

FYI, you can obtain the UI bundle from a deploy preview in any Antora environment by appending dist/ui-bundle.zip to the deploy preview URL. For example:

https://deploy-preview-127--cb-docs-ui.netlify.app/dist/ui-bundle.zip

You can either set that value in the playbook or pass it to Antora using the --ui-bundle-url CLI option.

This allows you to test a change to the UI with the real content before merging into the mainline.

**Pull request merged**